types/logger.d.ts

```diff
@@ -0,0 +1,39 @@
+/// <reference types="node" />
+
+import * as http from 'http'
+import * as net from 'net'
+
+export default interface FastifyLoggerOptions<
+  HttpRequest extends http.IncomingMessage = http.IncomingMessage,
+  HttpResponse extends http.OutgoingMessage = http.OutgoingMessage
+> {
+  serializers: {
+    req: (req: HttpRequest) => {
+      method: HttpRequest['method'],
+      url: HttpRequest['url'],
+      version: http.IncomingHttpHeaders['accept-version'],
+      hostname: any, // not on http.IncomingMessage
+      remoteAddress: any, // not on http.IncomingMessage
+      remotePort: net.Socket['remotePort']
+    },
+    err: (err: Error) => {
+      type: string;
+      message: string;
+      stack: string;
+      [key: string]: any;
+    },
+    res: (res: HttpResponse) => {
+      statusCode: any // not on http.OutgoingMessage
+    }
+  };
+  info: WriteFn;
+  error: WriteFn;
+  debug: WriteFn;
+  fatal: WriteFn;
+  warn: WriteFn;
+  trace: WriteFn;
+  child: WriteFn;
+  genReqId?: string;
+}
+
+type WriteFn = (o: object) => void;
```

Comment:
Would it make more sense if this was:

```ts
type WriteFn = (...args: any[]) => void;
```

Since it's a pino logger you can pass multiple args? Or am I mistaken to believe this? For example, it should be valid to do:

```ts
fastify.log.info('here is your json %j', someJson)
```

Reply:
Yes, that is a valid log invocation @evanshortiss .

Reply:
The current types have two methods defined. I'm working on merging the two right now.
```typescript
declare type WriteFn = (msg:string, ...args: any[]) => void;
declare type WriteFn = (obj: {}, msg?: string, ...args: any[]) => void;
```
TypeScript says I can't do it this way so I'm working on it 😅